### ISSUE TITLE
logger: use logging format interpolation

### ISSUE THREAD

**Issue by ghost** [iterative/dvc#1843]
E.g.

```
logger.debug("Saving {} to {}.".format(path_info, to_info))
```
should be
```
logger.debug("Saving '%s' to '%s'.", path_info, to_info)
```

~~As suggested by @ei-grad in this PR: https://github.com/iterative/dvc/pull/1822~~

~~Reference: https://docs.python.org/3/howto/logging.html#optimization~~

~~It needs to support braces `{}` formatting, you can use a log adapter for Python 2.7:~~
<s>```python
import logging

class Message(object):
    def __init__(self, fmt, args):
        self.fmt = fmt
        self.args = args

    def __str__(self):
        return self.fmt.format(*self.args)

class StyleAdapter(logging.LoggerAdapter):
    def __init__(self, logger, extra=None):
        super(StyleAdapter, self).__init__(logger, extra or {})

    def log(self, level, msg, *args, **kwargs):
        if self.isEnabledFor(level):
            msg, kwargs = self.process(msg, kwargs)
            self.logger._log(level, Message(msg, args), (), **kwargs)
```</s>

**Comment by shcheklein** [comment]
Consider introducing actual laziness - https://stackoverflow.com/questions/4148790/lazy-logger-message-string-evaluation . I'm not sure though it can be done w/o introducing some ugly wrapper.

**Comment by mroutis** [comment]
Closed by #3178

**Comment by efiop** [comment]
@mroutis Not closed, as there are plenty of placed that still `.format()` strings. Hence why I didn't include "Closing" into that PR. 🙂  Reopening.
